Sources/NIO/EventLoopFuture.swift

```diff
@@ -815,6 +815,33 @@ extension EventLoopFuture {
     }
 }
 
+extension EventLoopFuture {
+    /// Returns a new `EventLoopFuture` that fires only when this `EventLoopFuture` and
+    /// all the provided futures complete. It then provides the result of folding the value of this
+    /// `EventLoopFuture` with the values of all the provided futures.
+    ///
+    /// This function is suited when you have APIs that already know how to return `EventLoopFuture`s.
+    ///
+    /// The returned `EventLoopFuture` will fail with the first error encountered, as soon as this
+    /// `EventLoopFuture` or any of the provided futures fail: otherwise, it will succeed only when all
+    /// of them do.
```

Comment:
Ok, so let's address this comment to clarify exactly what will happen here. We should reword to:

> The returned `EventLoopFuture` will fail as soon as the a failure is encountered in any of the `futures` (or in this one). However, the failure will not occur until all preceding `EventLoopFuture`s have completed. At the point the failure is encountered, all subsequent `EventLoopFuture` objects will no longer be waited for. This function therefore fails *fast*: once a failure is encountered, it will immediately fail the overall `EventLoopFuture`.